File: Color-Palette-Manager/artifacts/clinderma/ml-training/dataset/organize.py
import os
import shutil
import re

dataset_dir = r"c:\Users\lenovo\Downloads\clinderma\Color-Palette-Manager\artifacts\clinderma\ml-training\dataset"

def organize_images():
    for root, dirs, files in os.walk(dataset_dir):
        if root == dataset_dir: continue # skip the top level since we are dealing with subfolders right now
        
        for file in files:
            if not file.lower().endswith(('.png', '.jpg', '.jpeg')): continue
            
            # Extract class name from something like "acne_vulgaris271.jpg"
            # We strip digits, underscores at the end, etc.
            # Example: "melanoma30_4.jpg" -> "melanoma"
            name_no_ext = os.path.splitext(file)[0]
            # remove trailing numbers and underscores
            class_name = re.sub(r'[\d_]+$', '', name_no_ext).strip('_').lower()
            
            # Fallback if class_name ends up empty
            if not class_name:
                class_name = "unknown"
                
            target_dir = os.path.join(dataset_dir, class_name)
            if not os.path.exists(target_dir):
                os.makedirs(target_dir)
                
            src_path = os.path.join(root, file)
            dst_path = os.path.join(target_dir, file)
            
            print(f"Moving {file} -> {class_name}/")
            shutil.move(src_path, dst_path)

if __name__ == "__main__":
    organize_images()
    print("Done organizing.")

File: Color-Palette-Manager/artifacts/clinderma/ml-training/dataset/organize.js
const fs = require('fs');
const path = require('path');

const datasetDir = __dirname;

function organizeImages() {
    const entries = fs.readdirSync(datasetDir, { withFileTypes: true });

    for (const entry of entries) {
        if (entry.isDirectory()) {
            const currentDirPath = path.join(datasetDir, entry.name);
            const subEntries = fs.readdirSync(currentDirPath, { withFileTypes: true });

            for (const subEntry of subEntries) {
                if (subEntry.isFile() && /\.(png|jpg|jpeg)$/i.test(subEntry.name)) {
                    let nameNoExt = path.parse(subEntry.name).name;
                    // Regex removes trailing digits/underscores (e.g. acne29_1 -> acne)
                    let className = nameNoExt.replace(/[\d_]+$/, '').replace(/^_+|_+$/g, '').toLowerCase();

                    if (!className) className = 'unknown';

                    const targetDir = path.join(datasetDir, className);
                    if (!fs.existsSync(targetDir)) {
                        fs.mkdirSync(targetDir);
                    }

                    const srcPath = path.join(currentDirPath, subEntry.name);
                    const dstPath = path.join(targetDir, subEntry.name);

                    console.log(`Moving ${subEntry.name} -> ${className}/`);
                    fs.renameSync(srcPath, dstPath);
                }
            }
        } else if (entry.isFile() && /\.(png|jpg|jpeg)$/i.test(entry.name)) {
            // Root level files
            let nameNoExt = path.parse(entry.name).name;
            let className = nameNoExt.replace(/[\d_]+$/, '').replace(/^_+|_+$/g, '').toLowerCase();

            if (!className) className = 'unknown';

            const targetDir = path.join(datasetDir, className);
            if (!fs.existsSync(targetDir)) {
                fs.mkdirSync(targetDir);
            }

            const srcPath = path.join(datasetDir, entry.name);
            const dstPath = path.join(targetDir, entry.name);

            console.log(`Moving ${entry.name} -> ${className}/`);
            fs.renameSync(srcPath, dstPath);
        }
    }
}

organizeImages();
console.log("Done organizing.");
